mlir/tools/rocmlir-gen/rocmlir-gen.cpp

```diff
@@ -2006,9 +2005,13 @@ static void emitMemcpy(OpBuilder &b, Value src, Value dst) {
   Value dstFlat = makeNDMemRef(b, dst, 1);
   auto srcFlatType = srcFlat.getType().cast<MemRefType>();
   auto dstFlatType = dstFlat.getType().cast<MemRefType>();
-  auto memcpyFunc = getMemcpyFuncDecl(module, srcFlatType, dstFlatType);
 
-  b.create<func::CallOp>(loc, memcpyFunc, ValueRange{srcFlat, dstFlat});
+  if (srcFlatType == dstFlatType) {
```

Comment:
The old code assumed/knew that there would be a type conversion here, and called getMemcpyFuncDecl() in all cases.  The new version accounts for cases without a conversion.